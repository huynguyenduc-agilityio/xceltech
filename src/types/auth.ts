export interface ILoginUser {
  email: string;
  password: string;
}

export interface IAuthUser {
  refresh: string;
  access: string;
  user: {
    id: string;
    email: string;
  };
}

export interface IRegisterUser extends ILoginUser {
  firstName: string;
  lastName: string;
  phone: string;
  confirmPassword: string;
  isAgreeTerms?: boolean;
  isReceiveNewsletters?: boolean;
}

export interface IActiveUser {
  uidb64: string | null;
  token: string | null;
}
