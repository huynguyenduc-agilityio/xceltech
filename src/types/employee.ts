export interface IEmployeeFamilyInfo {
  id?: string;
  fullName: string;
  relationship: string;
  phone: string;
  address: string;
}

export interface IEmployeeGuarantor {
  id?: string;
  name: string;
  job: string;
  phone: string;
}

export interface IEmployeeFinancialInfo {
  id?: string;
  bankName: string;
  accountNumber: number;
  accountName: string;
}

export interface IEmployeeEducationInfo {
  id?: string;
  name: string;
  department: string;
  course: string;
  location: string;
  startDate?: Date;
  endDate?: Date;
  description: string;
}

export type DocumentsForm = {
  offerLetter?: File | null;
  birthCertificate?: File | null;
  guarantorForm?: File | null;
  degreeCertificate?: File | null;
};

export type Documents = {
  documentFile: string;
  documentType: string;
};
