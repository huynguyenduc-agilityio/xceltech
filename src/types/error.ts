export type ErrorType = {
  detail: string;
};
