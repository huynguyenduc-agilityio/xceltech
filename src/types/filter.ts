export interface FilterCriteria {
  [key: string]: string | number | boolean | string[];
}
