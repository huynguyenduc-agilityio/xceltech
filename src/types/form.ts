export interface UploadFileForm {
  [key: string]: File | null;
}

export type TField<T> = {
  name: keyof T;
  label: string;
  placeholder: string;
  min?: string;
  max?: string;
  rules: {
    required: string;
    pattern: {
      value: RegExp;
      message: string;
    };
  };
  ariaLabel: string;
};

export enum MutationType {
  Create = 'create',
  Edit = 'edit',
}
