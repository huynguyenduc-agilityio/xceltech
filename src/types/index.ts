export * from './status';
export * from './employee';
export * from './table';
export * from './auth';
export * from './form';
export * from './tab';
export * from './user';
export * from './queryKey';
export * from './leave';
export * from './filter';
export * from './error';
export * from './notification';
export * from './profile';
