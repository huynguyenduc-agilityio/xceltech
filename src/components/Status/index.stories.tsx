import type { Meta, StoryObj } from '@storybook/react';

// Types
import { StatusLeave } from '@/types';

// Components
import Status from '.';

const meta = {
  title: 'Components/Status',
  component: Status,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
} satisfies Meta<typeof Status>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {},
};

export const Approved: Story = {
  args: {
    type: StatusLeave.Approved,
  },
};

export const Rejected: Story = {
  args: {
    type: StatusLeave.Rejected,
  },
};
