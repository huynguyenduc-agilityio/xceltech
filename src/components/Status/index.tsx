// Icons
import { Check, X } from 'lucide-react';

// Types
import { StatusLeave } from '@/types';

// Utils
import { cn } from '@/utils';

export interface StatusProps {
  type?: StatusLeave;
}

const Status = ({ type = StatusLeave.Pending }: StatusProps) => {
  const statusClasses = {
    [StatusLeave.Pending]: 'bg-gray-500',
    [StatusLeave.Approved]: 'bg-green-primary',
    [StatusLeave.Rejected]: 'bg-red-deep',
  };

  return (
    <div
      className={cn(
        'flex items-center w-28 justify-center gap-1 px-2.5 py-1 rounded-full border text-white',
        statusClasses[type],
      )}
    >
      {type === StatusLeave.Approved && <Check size={18} />}
      {type === StatusLeave.Rejected && <X size={18} />}
      <span className={cn('text-base font-bold capitalize')}>{type}</span>
    </div>
  );
};

export default Status;
