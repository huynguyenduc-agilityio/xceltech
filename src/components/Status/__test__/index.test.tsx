import { render, screen } from '@testing-library/react';

// Types
import { StatusLeave } from '@/types';

// Component
import Status from '..';

describe('Status Component', () => {
  it('renders with default status Pending', () => {
    render(<Status />);

    expect(screen.getByText('Pending')).toBeInTheDocument();
  });

  it('renders with Approved status', () => {
    render(<Status type={StatusLeave.Approved} />);

    expect(screen.getByText('Approved')).toBeInTheDocument();
  });

  it('renders with Rejected status', () => {
    render(<Status type={StatusLeave.Rejected} />);

    expect(screen.getByText('Rejected')).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = render(<Status />);

    expect(container).toMatchSnapshot();
  });
});
