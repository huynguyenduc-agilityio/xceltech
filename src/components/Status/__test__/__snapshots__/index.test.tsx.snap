// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status Component should match snapshot 1`] = `
<div>
  <div
    class="flex items-center w-28 justify-center gap-1 px-2.5 py-1 rounded-full border text-white bg-gray-500"
  >
    <span
      class="text-base font-bold capitalize"
    >
      Pending
    </span>
  </div>
</div>
`;
